examples_tests/13.BRDF_Explorer/createDerivMapFromBumpMap.cpp

```diff
@@ -0,0 +1,56 @@
+#include <irrlicht.h>
+#include "createComputeShader.h"
+#include "../source/Irrlicht/COpenGL2DTexture.h"
+#include "../source/Irrlicht/COpenGLDriver.h"
+#include "CComputeShaderManager.h"
+
+using namespace irr;
+
+video::IVirtualTexture* createDerivMapFromBumpMap(video::IVirtualTexture* _bumpMap, IrrlichtDevice* _device, float _heightFactor)
+{
+    asset::IAssetManager& assetMgr = _device->getAssetManager();
+    asset::IAssetLoader::SAssetLoadParams lparams;
+
+    video::IVideoDriver* driver = _device->getVideoDriver();
+
+    const uint32_t* derivMap_sz = _bumpMap->getSize();
+    video::ITexture* derivMap = driver->createGPUTexture(video::ITexture::ETT_2D, derivMap_sz, std::log2(std::max(derivMap_sz[0], derivMap_sz[1])), asset::EF_R8G8_SNORM);
+
+    {
+        video::STextureSamplingParams params;
+        params.UseMipmaps = 0;
+        params.MaxFilter = params.MinFilter = video::ETFT_LINEAR_NO_MIP;
+        params.TextureWrapU = params.TextureWrapV = video::ETC_CLAMP_TO_EDGE;
```

Comment:
you need a choice (argument to the function) whether this will simulate repeating heightfield (`ETC_REPEAT`) or not

Reply:
@Crisspl ?

Reply:
@Crisspl ?

Reply:
resolved